src/MeoAssistant/AsstUtils.hpp

```diff
@@ -32,63 +32,102 @@ namespace asst::utils
-        _string_replace_all(str, old_value, new_value);
+        std::basic_string<char_t> str = src;
+        _string_replace_all(str, replace_pair);
         return str;
     }
 
-    inline std::string string_replace_all_batch(
-        const std::string& src, std::initializer_list<std::pair<std::string_view, std::string_view>>&& replace_pairs)
+    template <typename char_t>
+    inline std::basic_string<char_t> string_replace_all(
+        const std::basic_string<char_t>& src, std::initializer_list<pair_of_string_view<char_t>>&& replace_pairs)
     {
-        std::string str = src;
+        std::basic_string<char_t> str = src;
         for (const auto& [old_value, new_value] : replace_pairs) {
             _string_replace_all(str, old_value, new_value);
         }
         return str;
     }
 
-    template <typename map_t>
-    inline std::string string_replace_all_batch(const std::string& src, const map_t& replace_pairs)
-    requires std::derived_from<typename map_t::value_type::first_type, std::string> &&
-             std::derived_from<typename map_t::value_type::second_type, std::string>
+    template <typename char_t, typename map_t>
+    requires std::derived_from<typename map_t::value_type::first_type, std::basic_string<char_t>> &&
+             std::derived_from<typename map_t::value_type::second_type, std::basic_string<char_t>>
+    [[deprecated]] inline std::basic_string<char_t> string_replace_all(const std::basic_string<char_t>& src,
+                                                                       const map_t& replace_pairs)
     {
-        std::string str = src;
+        std::basic_string<char_t> str = src;
         for (const auto& [old_value, new_value] : replace_pairs) {
             _string_replace_all(str, old_value, new_value);
         }
         return str;
     }
 
-    inline std::vector<std::string> string_split(const std::string& str, const std::string& delimiter)
+    // 以低内存开销拆分一个字符串；注意当 src 析构后，返回值将失效
+    template <typename str_t, typename char_t = typename str_t::value_type, typename delimiter_t>
+    requires std::convertible_to<str_t, std::basic_string_view<char_t>> &&
+             std::convertible_to<delimiter_t, std::basic_string_view<char_t>>
+    inline std::vector<std::basic_string_view<char_t>> string_split(const str_t& src,
+                                                                    delimiter_t delimiter)
     {
-        std::string::size_type pos1 = 0;
-        std::string::size_type pos2 = str.find(delimiter);
-        std::vector<std::string> result;
+        std::basic_string_view<char_t> delimiter_view = delimiter;
+        std::basic_string_view<char_t> str = src;
+        typename std::basic_string<char_t>::size_type pos1 = 0;
+        typename std::basic_string<char_t>::size_type pos2 = str.find(delimiter_view);
+        std::vector<std::basic_string_view<char_t>> result;
 
-        while (std::string::npos != pos2) {
+        while (pos2 != str.npos) {
```

Comment:
要不
```suggestion
        while (pos2 != str::npos) {
```
前面好像也有

还有这个 `requires` 限制不知道是不是有必要 🤔